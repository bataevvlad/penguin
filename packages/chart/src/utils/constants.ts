export const chartPaddingHorizontal = 40;
export const chartPaddingVertical = 40;
export const defaultChartHeightDivider = 4;
export const COLORS = [
  '#E0F5E9',
  '#9BE1DA',
  '#56CDCB',
  '#3B8B8E',
  '#EF1E1C',
  '#F43B1D',
  '#F9571D',
  '#FF8A57',
];
