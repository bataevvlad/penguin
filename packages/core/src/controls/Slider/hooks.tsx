import {useCallback, useRef, useMemo, MutableRefObject} from 'react';
import {Animated, I18nManager, LayoutChangeEvent} from 'react-native';

import {clamp} from './helpers';

export const useLowHigh = (
  lowProp: number | undefined,
  highProp: number | undefined,
  min: number,
  max: number,
  step: number,
) => {
  const validLowProp = lowProp === undefined ? min : clamp(lowProp, min, max);
  const validHighProp =
    highProp === undefined ? max : clamp(highProp, min, max);
  const inPropsRef = useRef({
    low: validLowProp,
    high: validHighProp,
    step,
    // These 2 fields will be overwritten below.
    min: validLowProp,
    max: validHighProp,
  });
  const {low: lowState, high: highState} = inPropsRef.current;
  const inPropsRefPrev = {lowPrev: lowState, highPrev: highState};

  // Props have higher priority.
  // If no props are passed, use internal state variables.
  const low = clamp(lowProp === undefined ? lowState : lowProp, min, max);
  const high = clamp(highProp === undefined ? highState : highProp, min, max);

  // Always update values of refs so pan responder will have updated values
  Object.assign(inPropsRef.current, {low, high, min, max});

  const setLow = (value: number) => (inPropsRef.current.low = value);
  const setHigh = (value: number) => (inPropsRef.current.high = value);
  return {inPropsRef, inPropsRefPrev, setLow, setHigh};
};

/**
 * Sets the current value of widthRef and calls the callback with new width parameter.
 * @param widthRef
 * @param callback
 * @returns {function({nativeEvent: *}): void}
 */
export const useWidthLayout = (
  widthRef: MutableRefObject<number>,
  callback?: (width: number) => void,
) => {
  return useCallback(
    (event: LayoutChangeEvent) => {
      const {
        layout: {width},
      } = event.nativeEvent;
      const {current: w} = widthRef;
      if (w !== width) {
        widthRef.current = width;
        if (callback) {
          callback(width);
        }
      }
    },
    [callback, widthRef],
  );
};

interface IProps {
  low: number;
  high: number;
  min: number;
  max: number;
  step: number;
}

export const useSelectedRail = (
  inPropsRef: MutableRefObject<IProps>,
  containerWidthRef: MutableRefObject<number>,
  thumbWidth: number,
  disableRange: boolean,
) => {
  const {current: left} = useRef(new Animated.Value(0));
  const {current: right} = useRef(new Animated.Value(0));
  const update = useCallback(() => {
    const {low, high, min, max} = inPropsRef.current;
    const {current: containerWidth} = containerWidthRef;
    const fullScale = (max - min) / (containerWidth - thumbWidth);
    const leftValue = (low - min) / fullScale;
    const rightValue = (max - high) / fullScale;
    left.setValue(disableRange ? 0 : leftValue);
    right.setValue(
      disableRange ? containerWidth - thumbWidth - leftValue : rightValue,
    );
  }, [inPropsRef, containerWidthRef, disableRange, thumbWidth, left, right]);
  const styles = useMemo(
    () => ({
      left: I18nManager.isRTL ? right : left,
      right: I18nManager.isRTL ? left : right,
    }),
    [left, right],
  );
  return [styles, update] as const;
};
