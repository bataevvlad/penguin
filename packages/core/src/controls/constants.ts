enum ControlType {
  checkBox = 'checkBox',
  radio = 'radio',
}

export default ControlType;
