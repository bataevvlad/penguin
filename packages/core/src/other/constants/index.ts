import {LABELS} from './LABELS';

export {LABELS};
