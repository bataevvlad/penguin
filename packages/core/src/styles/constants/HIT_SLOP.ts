import px from '../utils/px';

const HIT_SLOP = {
  small: {top: px(10), bottom: px(10), left: px(10), right: px(10)},
  medium: {top: px(20), bottom: px(20), left: px(20), right: px(20)},
  large: {top: px(30), bottom: px(30), left: px(30), right: px(30)},
};

export default HIT_SLOP;
