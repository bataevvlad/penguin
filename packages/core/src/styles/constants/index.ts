import HIT_SLOP from './HIT_SLOP';

export * from './colors';
export * from './defaultSpaces';

export * from './theme';

export {HIT_SLOP};
