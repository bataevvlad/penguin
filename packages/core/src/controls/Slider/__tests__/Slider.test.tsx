import {act, fireEvent, render} from '@testing-library/react-native';

import Slider from '../Slider';
import {LABELS} from '../../../other/constants';
import HIT_SLOP from '../../../styles/constants/HIT_SLOP';

const eventMock = {};
const trueFunc = () => true;
const falseFunc = () => false;

describe.skip('Slider', () => {
  it('should renders correctly by default', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={0} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    act(() => panHandler.props.onMoveShouldSetPanResponder({}, {dx: -10}));
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with more props', function () {
    const {getByLabelText, toJSON} = render(
      <Slider
        min={0}
        max={100}
        step={1}
        minRange={10}
        low={20}
        high={60}
        disableRange={true}
        hitSlop={HIT_SLOP.small}
      />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    act(() => panHandler.props.onMoveShouldSetPanResponder({}, {dx: -10}));
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly panHandler', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={0} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);

    act(() => {
      panHandler.props.onStartShouldSetPanResponderCapture(falseFunc);
      panHandler.props.onMoveShouldSetPanResponderCapture(falseFunc);
      panHandler.props.onPanResponderTerminationRequest(falseFunc);
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 33,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderMove(eventMock, {moveX: 20});
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly disabled', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={0} max={100} step={1} disableRange={true} disabled={true} />,
    );
    const panHandler = getByLabelText(LABELS.slider);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 33,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
    });
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly updateThumbs', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={20} max={100} step={1} disableRange={true} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    const layout = getByLabelText(LABELS.sliderLayoutLowThumb);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 33,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderMove(eventMock, {moveX: 40});
    });
    expect(toJSON()).toMatchSnapshot();
    act(() => {
      fireEvent(layout, 'layout', {
        nativeEvent: {layout: {width: 100}},
      });
    });
  });
  it('should renders correctly with numberActiveTouches', function () {
    const {getByLabelText, toJSON} = render(
      <Slider
        min={0}
        max={100}
        step={1}
        minRange={10}
        low={20}
        high={60}
        disableRange={true}
        hitSlop={HIT_SLOP.small}
      />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    act(() =>
      panHandler.props.onPanResponderGrant({}, {numberActiveTouches: 2}),
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly updateThumbs high -> low', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={20} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    const layout = getByLabelText(LABELS.sliderLayoutLowThumb);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: -10,
            locationY: 5,
            pageX: 200,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderMove(eventMock, {moveX: 170});
    });
    expect(toJSON()).toMatchSnapshot();
    act(() => {
      fireEvent(layout, 'layout', {
        nativeEvent: {layout: {width: 100}},
      });
    });
  });
  it('should renders correctly high <- low', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={20} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    const layout = getByLabelText(LABELS.sliderLayoutLowThumb);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 200,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderMove(eventMock, {moveX: 175});
    });
    expect(toJSON()).toMatchSnapshot();
    act(() => {
      fireEvent(layout, 'layout', {
        nativeEvent: {layout: {width: 100}},
      });
    });
  });
  it('should renders correctly onPanResponderRelease', function () {
    const {getByLabelText, toJSON} = render(
      <Slider min={20} max={100} step={1} />,
    );
    const panHandler = getByLabelText(LABELS.slider);
    const layout = getByLabelText(LABELS.sliderLayoutLowThumb);

    act(() => {
      panHandler.props.onPanResponderTerminate(trueFunc);
      panHandler.props.onShouldBlockNativeResponder(trueFunc);
      panHandler.props.onMoveShouldSetPanResponder(eventMock, {dx: 10, dy: 0});
      panHandler.props.onPanResponderGrant(
        {
          nativeEvent: {
            locationX: 21,
            locationY: 5,
            pageX: 200,
            pageY: 152,
            target: 723,
          },
        },
        {dx: 10, dy: 0},
      );
      panHandler.props.onPanResponderRelease();
      panHandler.props.onPanResponderMove(eventMock, {moveX: 175});
    });
    expect(toJSON()).toMatchSnapshot();
    act(() => {
      fireEvent(layout, 'layout', {
        nativeEvent: {layout: {width: 100}},
      });
    });
  });
});
