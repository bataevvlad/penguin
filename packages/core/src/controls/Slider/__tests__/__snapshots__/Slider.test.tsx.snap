// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should renders correctly by default 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": -36,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly disabled 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": -36,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly high <- low 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": -22.05,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": -22.05,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 45,
            "borderWidth": 3,
            "height": 45,
            "width": 45,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": -36,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly onPanResponderRelease 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": -22.05,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": -22.05,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": -36,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly panHandler 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": -10.08,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": -10.080000000000002,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 45,
            "borderWidth": 3,
            "height": 45,
            "width": 45,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": -36,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly updateThumbs 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": -36,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 45,
            "borderWidth": 3,
            "height": 45,
            "width": 45,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly updateThumbs high -> low 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      collapsable={false}
      style={
        {
          "position": "absolute",
          "transform": [
            {
              "translateX": -36,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 45,
            "borderWidth": 3,
            "height": 45,
            "width": 45,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 45,
          "left": 45,
          "right": 45,
          "top": 45,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly with more props 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": -28.8,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": -7.2,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Slider should renders correctly with numberActiveTouches 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "minHeight": 48,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "bottom": 0,
            "flexDirection": "row",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "marginHorizontal": 18,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "#9BA1B0",
            "borderRadius": 3,
            "flex": 1,
            "height": 6,
          }
        }
      />
      <View
        collapsable={false}
        style={
          {
            "left": 0,
            "position": "absolute",
            "right": -28.8,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#2B78EE",
              "borderRadius": 3,
              "height": 6,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="sliderLayoutLowThumb"
      collapsable={false}
      style={
        {
          "transform": [
            {
              "translateX": -7.2,
            },
          ],
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FFF",
            "borderColor": "#9BA1B0",
            "borderRadius": 36,
            "borderWidth": 3,
            "height": 36,
            "width": 36,
          }
        }
      />
    </View>
    <View
      accessibilityLabel="slider"
      collapsable={false}
      hitSlop={
        {
          "bottom": 15,
          "left": 15,
          "right": 15,
          "top": 15,
        }
      }
      onMoveShouldSetPanResponder={[Function]}
      onMoveShouldSetPanResponderCapture={[Function]}
      onPanResponderGrant={[Function]}
      onPanResponderMove={[Function]}
      onPanResponderRelease={[Function]}
      onPanResponderTerminate={[Function]}
      onPanResponderTerminationRequest={[Function]}
      onShouldBlockNativeResponder={[Function]}
      onStartShouldSetPanResponderCapture={[Function]}
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
